src/operations/insert.ts

```diff
@@ -67,63 +69,30 @@ export class InsertOneOperation extends CommandOperation<InsertOneOptions, Inser
     const doc = this.doc;
     const options = { ...this.options, ...this.bsonOptions };
 
-    if (Array.isArray(doc)) {
-      return callback(
-        MongoError.create({ message: 'doc parameter must be an object', driver: true })
-      );
-    }
+    // Final options for retryable writes
+    // let finalOptions = Object.assign({}, options);
+    // finalOptions = applyRetryableWrites(finalOptions, coll.s.db);
```

Comment:
Do retryable writes get handled elsewhere?

Reply:
Yeah, they are handled in [executeOperation](https://github.com/mongodb/node-mongodb-native/blob/master/src/operations/execute_operation.ts#L209). We were actually out of [spec compliance](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst#retrywrites) by allowing it to be set at the operation level.

Reply:
Ah yep, looks like it.